### PR TITLE
React Stepper slider animations

### PR DESCRIPTION
**Pull request by bobwise**
## What does this PR do?
Add CSS animation for React Stepper

### Associated Issue 
https://github.com/sparkdesignsystem/spark-design-system/issues/1535

### Code
 - [x] Build Component in Spark React

### Accessibility
- [ ] New changes abide by [accessibility requirements](https://sparkdesignsystem.com/docs/accessibility)

### Browser Testing (current version and 1 prior)
  - [X] Google Chrome - Current
  - [X] Google Chrome - 1 Prior
  - [x] Google Chrome (Mobile) - Current
  - [ ] Google Chrome (Mobile) - 1 Prior (version 74)
  - [x] Mozilla Firefox - Current
  - [x] Mozilla Firefox - 1 Prior
  - [x] Mozilla Firefox (Mobile) - Current (version 18 on iOS, version 68 Android not tested)
  - [ ] Mozilla Firefox (Mobile) - 1 Prior (version 17 on iOS and/or version 68 on Android)
  - [x] Microsoft Internet Explorer 11 (only this specific version of IE)
  - [x] Microsoft Edge - Current
  - [x] Microsoft Edge - 1 Prior
  - [x] Apple Safari - Current
  - [x] Apple Safari - 1 Prior
  - [x] Apple Safari (Mobile) - Current
  - [x] Apple Safari (Mobile) - 1 Prior

### Design Review
 - [X] Design reviewed and approved

### CODE REVIEW

File: packages/spark/components/_stepper.scss [hunk 1]
```diff
@@ -40,6 +35,7 @@
 .sprk-c-Stepper__slider--active .sprk-c-Stepper__step-icon,
 .sprk-c-Stepper__slider--active .sprk-c-Stepper__step-description {
   opacity: 1;
+  transition: opacity $sprk-stepper-slider-text-fade-in-timing ease 0.2s;
```
Comment (13protons) [inline packages/spark/components/_stepper.scss]: @afebbraro do you see any issue with changing this transition to *only* apply to the active slider? The point here is to fade in, but not out. If you look at the way vanilla operates, the text and title of the stepper only ever **fade-in**, and so this change to the css technically does not change anything about how that interaction is rendered. If anything there's now just fewer unintended side effects.

File: src/react/projects/spark-react/src/SprkStepper/components/SprkStepperSlider/SprkStepperSlider.js [hunk 2]
```diff
@@ -1,8 +1,31 @@
...
 class SprkStepperSlider extends Component {
+  constructor(props) {
+    super(props);
+
+    this.sliderRef = React.createRef();
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.sliderStyle.top != this.props.sliderStyle.top){
+      this.sliderRef.current.classList.remove('sprk-c-Stepper__slider--active');
```
Comment (13protons) [inline src/react/projects/spark-react/src/SprkStepper/components/SprkStepperSlider/SprkStepperSlider.js]: can this cause/effect relationship be unit tested? Aka, change `top`, observe `--active` flicker? Not sure if that adds value now, but would stop it from changing in the future.

File: packages/spark/components/_stepper.scss (hunk 1 above)
Comment (afebbraro) [reply packages/spark/components/_stepper.scss]: Hmm, I actually didn't write the sprk-c-Stepper__slider--active class name and I think I had it on the selected step originally. But, scoping the transition in this way seems fine to do and like you said doesn't change anything about how it works currently. So, if it solves some weird React bug then I don't see harm in doing this, especially like you said, if it adds more predicability. Let's do it.

Comment (afebbraro) [reply packages/spark/components/_stepper.scss]: Oh! But can you make the delay a var? @bobwise. The hard-coded 0.2s should be placed as var in our settings file.

File: src/patterns/components/stepper/collection.yaml [hunk 3]
```diff
@@ -50,7 +53,7 @@ variableTable:
   $sprk-stepper-dark-icon-color-selected:
     default: $sprk-white
     description: The color of the step icon when the step is selected and the stepper has a dark background (sprk-c-Stepper--has-dark-bg).
-  $sprk-stepper-dark-icon-color-hoverg:
+  $sprk-stepper-dark-icon-color-hover:
```
Comment (afebbraro) [inline src/patterns/components/stepper/collection.yaml]: Good spelling catch!

File: src/react/projects/spark-react/src/SprkStepper/components/SprkStepperSlider/SprkStepperSlider.js (hunk 2 above)
Comment (RCopeland) [reply src/react/projects/spark-react/src/SprkStepper/components/SprkStepperSlider/SprkStepperSlider.js]: i think componentWIllReceiveProps is deprecated also

**Comment by afebbraro** [comment]
Approved and asked Bob to update the animations issue bug with it also being an issue in Vanilla.

**Pull request merged**